app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -386,10 +386,13 @@ def remove_middleware_datasource(ems_ref)
     end
 
     def emit_middleware_notification(type, op_name, op_arg, ems_ref)
+      mw_deployment = MiddlewareDeployment.find_by_ems_ref(ems_ref)
+      mw_server = mw_deployment.nil? ? MiddlewareServer.find_by_ems_ref(ems_ref) :
```

Comment:
@ammendonca - it looks like you know the class when you call the ```emit_middleware_notification``` method, right? You can just pass in the class name as an argument.